src/survey/controllers/create_survey_controller.py

```diff
@@ -1,49 +1,75 @@
+import json
+
 from api.utils import failure_response
 from api.utils import modify_attribute
 from api.utils import success_response
 from match import match_status
 from match.models import Match
-from person.models import Person
 from rest_framework import status
-from survey.constants import RATINGS
+from survey import constants
 from survey.models import Survey
 
 
 # TODO(@chalo2000) Convert to celery task
 class CreateSurveyController:
-    def __init__(self, request, data):
+    def __init__(self, request, data, match_id):
         self._request = request
         self._data = data
+        self._match_id = match_id
 
     def process(self):
         # Verify that all required fields are provided
         did_meet = self._data.get("did_meet")
-        explanation = self._data.get("explanation")
-        completed_match_id = self._data.get("completed_match_id")
-        if explanation is None:
-            return failure_response(
-                "POST body is misformatted", status.HTTP_400_BAD_REQUEST
-            )
+        if did_meet is None:
+            return failure_response("did_meet", status.HTTP_400_BAD_REQUEST)
 
-        # Get optional fields
+        # Get optional fields based on did_meet
+        did_meet_reason = self._data.get("did_meet_reason")
+        did_not_meet_reasons = self._data.get("did_not_meet_reasons")
```

Comment:
You can get `did_not_meet_reasons` and `rating` inside the if statements since they're only used in specific conditions.

Reply:
nevermind I see them being used at the bottom of this